### PR TITLE
history / version

### PR DESCRIPTION
**Pull request by liampauling**

### CODE REVIEW

File: betfairlightweight/streaming/betfairstream.py [hunk 1]
```diff
@@ -48,6 +48,7 @@ def __init__(
         self.datetime_last_received = None
 
         self._socket = None
```
[the remark this answers is not in the document]
Comment (liampauling) [reply betfairlightweight/streaming/betfairstream.py:50]: @petedmarsh Oh, actually `_socket` and `_socket_file` are the same, can we just have one?

[the remark this answers is not in the document]
Comment (petedmarsh) [reply betfairlightweight/streaming/betfairstream.py:50]: You need both I'm afraid :)

**Comment by liampauling** [comment]
@petedmarsh I am currently testing this branch, in regards to the socket code I am actually seeing an increase in latency compared to v2.18.0. Both instances doing the same thing, on the same VM / libraries etc, has me a bit baffled...

**Comment by petedmarsh** [comment]
That's interesting, how are you measuring the latency?

**Comment by petedmarsh** [comment]
OK so it's also worse for me on Linux, I ran the benchmark and tested on Windows, please revert that change. I believe I can make it work on Windows and Linux.

**Comment by petedmarsh** [comment]
Linux benchmark numbers for me:

```
                                                    Benchmarks, repeat=5, number=5
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
┃                                       Benchmark ┃ Min     ┃ Max     ┃ Mean    ┃ Min (+)         ┃ Max (+)         ┃ Mean (+)        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
│ Original Client Code vs socket.makefile version │ 1.160   │ 1.357   │ 1.204   │ 1.808 (-1.6x)   │ 1.829 (-1.3x)   │ 1.819 (-1.5x)   │
└─────────────────────────────────────────────────┴─────────┴─────────┴─────────┴─────────────────┴─────────────────┴─────────────────┘
```

**Comment by liampauling** [comment]
I don't understand why though, surely it should be quicker, might try and have a look as the latency distribution is weird

**Comment by petedmarsh** [comment]
@liampauling I believe I know why this was faster on Windows, slower on Linux (and OSX), and I believe I have a simple fix that makes an improvement cross-platform.

I have tested this on Windows and OSX, I can check Linux later, could you run this benchmark:

https://gist.github.com/petedmarsh/0a2775ec706156b892d40a67cb017bef

If you can verify it's positive for you too then I will raise the fix with a full explanation.

**Pull request merged**